File: Java/P4/irrgarten/irrgarten/src/irrgarten/CombatElement.java

package irrgarten;

/**
 *
 * @author isabel
 */
abstract class CombatElement {
    
    private float effect;
    private int uses;
    
    public CombatElement(float effect, int uses){
        this.effect = effect;
        this.uses = uses;
    }
    
    protected float produceEffect(){
        float result = 0;
        
        if (uses > 0){
            uses--;
            result = effect;
        }
        
        return result;
    }

    /**
     * @brief   Toma decisión de si un arma es descacartada
     * @return  true si descartado o false en caso contrario
     */
    public boolean discard(){
        return Dice.discardElement(uses);
    }
    
    @Override
    public String toString(){
        String cadena = Float.toString(effect) + ", " + 
                        Integer.toString(uses) + "]";
        return cadena;    
    };
}

File: Ruby/P4/Irrgarten/combat_element.rb
#encoding:utf-8

=begin

    Autor: Isabel Morro Tabares 

    Define los tipos de elementos para el combate

    
=end

require_relative 'Dice'

module Irrgarten
    class CombatElement

        def initialize(effect, uses)
            @effect = effect
            @uses = uses
        end

        def produce_effect
            resultado = 0
            if @uses > 0  
                @uses = @uses - 1
                resultado = @effect
            end
            resultado
        end

        def discard
            Dice.discard_element(@uses)
        end

        def to_s
         cadena = @effect.to_s 
         cadena += ", " + @uses.to_s 
         cadena += "]"
         cadena
        end

    end
end